### ISSUE TITLE
Add KV Command: Write Key/Value Pair on Namespace

### ISSUE THREAD

**Issue by ashleymichal** [cloudflare/wrangler#343]
As a Workers KV user, I should be able to set a key/value pair on my KV namespace from Wrangler by typing

``` sh
$ wrangler kv write myNamespace myKey=”someStringValue” --exp 2d --ttl 60s
```

**Comment by hughlang** [comment]
In the legacy docs, the process was to add the KV binding into the metadata.json file in your project.  Example from: https://developers.cloudflare.com/workers/api/resource-bindings/kv-namespaces/

```
{
  "body_part": "script",
  "bindings": [
    {
      "type": "kv_namespace",
      "name": "myNamespace",
      "namespace_id": ":namespace_id"
    }
  ]
}
```

However, this is overwritten every time you run `wrangler publish`. If wrangler did write these values successfully into metadata.json, would the KV bindings get set during the upload?

**Comment by steveklabnik** [comment]
@hughlang that's covered by https://github.com/cloudflare/wrangler/issues/92 which will be fixed in https://github.com/cloudflare/wrangler/pull/334

**Comment by EverlastingBugstopper** [comment]
This looks great!

Two things
1) For the long key example - how would wrangler know it's a path? Maybe we could do `@/path/to/file`?
2) `--exp` feels like a non-obvious abbreviation, maybe we do `--expiration` and `-E`?

**Comment by gabbifish** [comment]
Perhaps, to distinguish between uploading string values or values read from a file, we can explicitly state:
1. if a value is provided in quotation marks `""`, it will be read as a string.
2. if a value is provided without quotation marks, it will be treated as a file.

Other thoughts: we need to figure out what would happen if a user specified both --exp and --ttl, since these options conflict: https://workers.cloudflare.com/docs/reference/storage/expiring-keys/. I am onboard with having --ttl, by default, override --exp, given that --exp uses unix epoch values, which is slightly clunkier and probably not as easy to use as --ttl.

**Comment by gabbifish** [comment]
^ Actually, following up on this--we can just fallback to whatever the API returns if both `expiration` and `expiration_ttl` are specified as URL params.

**Comment by gabbifish** [comment]
UPDATE: my string idea above (use `""` to determine string vs filename) won't work--Clap will parse all quotation marks out of arguments. We could use avery's option? Or do we do something as explicit as 
```
$ wrangler kv write --key=gabbi --value=fisher
```
VS
```
$ wrangler kv write --key --file=a.txt
```
I also want to make sure this individual write usage is somewhat symmetric with our bulk write: 
https://github.com/cloudflare/wrangler/pull/445

**Comment by ashleymichal** [comment]
hmm. how about we make them ordered args, so 
```
$ wrangler kv write mykey myValue
```
and give a flag like `--file` if it's a file otherwise treat it like a string, so
```
$ wrangler kv write mykey myFile.txt --file
```
At least to get us past the first implementation.

**Comment by gabbifish** [comment]
^ I'll go with @ashleymichal's suggestion for now, and we can revisit when we have a monster day of wrangler kv usage discussions :)

**Comment by ashleymichal** [comment]
closed by #458
